extensions/ql-vscode/src/run-queries.ts

```diff
@@ -334,7 +334,11 @@ async function determineSelectedQuery(selectedResourceUri: vscode.Uri | undefine
   if (queryUri.scheme !== 'file') {
     throw new Error('Can only run queries that are on disk.');
   }
-  const queryPath = queryUri.fsPath;
+  const queryPath = queryUri.fsPath || '';
+
+  if (!queryPath.endsWith('.ql')) {
+    throw new Error('The selected resource is not a QL file.');
```

Comment:
```suggestion
    throw new Error('The selected resource is not a CodeQL query file; It should have the extension ".ql".');
```

Reply:
Or... maybe some variant of that. 
`The selected resource is not a CodeQL query file. (extension ".ql")`
`The selected resource is not a CodeQL query file. CodeQL queries have the extension ".ql".`
I mostly just want to avoid the confusion that `.qll` files are arguably, at least to a newcomer, a kind of QL file, in that they are QL libraries.